File: ion/src/com/koushikdutta/ion/gson/GsonObjectParser.java
package com.koushikdutta.ion.gson;

import com.google.gson.JsonObject;

import java.nio.charset.Charset;

/**
 * Created by koush on 6/23/14.
 */
public class GsonObjectParser extends GsonParser<JsonObject> {
    public GsonObjectParser() {
        super(JsonObject.class);
    }
    public GsonObjectParser(Charset charset) {
        super(JsonObject.class, charset);
    }
}

File: ion/src/com/koushikdutta/ion/gson/GsonObjectParser.kt
package com.koushikdutta.ion.gson

import com.google.gson.JsonObject
import java.nio.charset.Charset

/**
 * Created by koush on 6/23/14.
 */
class GsonObjectParser : GsonParser<JsonObject> {
    constructor() : super(JsonObject::class.java) {}
    constructor(charset: Charset?) : super(JsonObject::class.java, charset!!)
}

class GsonObjectSerializer : GsonSerializer<JsonObject?>()